sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts

```diff
@@ -27,28 +32,47 @@ export class AssessmentAESTController extends BaseController {
     private readonly studentAppealService: StudentAppealService,
     private readonly studentAssessmentService: StudentAssessmentService,
     private readonly assessmentControllerService: AssessmentControllerService,
+    private readonly applicationExceptionService: ApplicationExceptionService,
   ) {
     super();
   }
 
   /**
-   * Method to get all requested assessments for a student
+   * Get all requests related to an assessments for a student
    * application, i.e, this will fetch all pending and denied
-   * student appeal for an application.
-   * @param applicationId, application number.
-   * @returns assessment requests for a student application.
+   * student appeals for an application or possible application
+   * exceptions that will prevent the assessment to proceed till
+   * they are approved, for instance, when a document is uploaded
+   * and need to be reviewed.
+   * @param applicationId application number.
+   * @returns assessment requests or exceptions for a student application.
    */
   @Get("application/:applicationId/requests")
   async getRequestedAssessmentSummary(
     @Param("applicationId") applicationId: number,
   ): Promise<RequestAssessmentSummaryAPIOutDTO[]> {
+    const applicationExceptions =
+      await this.applicationExceptionService.getExceptionsByApplicationId(
+        applicationId,
+        ApplicationExceptionStatus.Pending,
+        ApplicationExceptionStatus.Declined,
+      );
+    if (applicationExceptions?.length) {
```

Comment:
if there are no elements, will it be undefined or empty array?

Reply:
It will probably be undefined. Do you think it would be better to remove the `.length`, I guess?

Reply:
ya....because, if it's empty then, `.map` will manage it. right